ruby_event_store/lib/ruby_event_store/repositories/null.rb

```diff
@@ -0,0 +1,17 @@
+module RubyEventStore
+  module Repositories
+    class Null
+      def initialize(repository_to_mimic = InMemory.new)
+        @repository_to_mimic = repository_to_mimic
+      end
+
+      def method_missing(method_name, *_)
+        super unless respond_to_missing?(method_name)
+      end
+
+      def respond_to_missing?(method_name, include_private = false)
+        @repository_to_mimic.respond_to?(method_name, include_private)
+      end
+    end
+  end
+end
```

Comment:
I had to check in IRB how would that work.

```
irb(main):007:0* module RubyEventStore
irb(main):008:1>   module Repositories
irb(main):009:2>     class Null
irb(main):010:3>       def initialize(repository_to_mimic = InMemory.new)
irb(main):011:4>         @repository_to_mimic = repository_to_mimic
irb(main):012:4>       end
irb(main):013:3> 
irb(main):014:3*       def method_missing(method_name, *_)
irb(main):015:4>         super unless respond_to_missing?(method_name)
irb(main):016:4>       end
irb(main):017:3> 
irb(main):018:3*       def respond_to_missing?(method_name, include_private = false)
irb(main):019:4>         @repository_to_mimic.respond_to?(method_name, include_private)
irb(main):020:4>       end
irb(main):021:3>     end
irb(main):022:2>   end
irb(main):023:1> end
=> :respond_to_missing?
irb(main):024:0> RubyEvetStore::Repositories::Null
NameError: uninitialized constant RubyEvetStore
	from (irb):24
	from /Users/rupert/.rbenv/versions/2.2.8/bin/irb:11:in `<main>'
irb(main):025:0> RubyEventStore::Repositories::Null
=> RubyEventStore::Repositories::Null
irb(main):026:0> RubyEventStore::Repositories::Null.new
=> #<RubyEventStore::Repositories::Null:0x007f90741ebf48 @repository_to_mimic=#<InMemory:0x007f90741ebf20>>
irb(main):027:0> RubyEventStore::Repositories::Null.new.asd
=> nil
irb(main):028:0> RubyEventStore::Repositories::Null.new.def
NoMethodError: undefined method `def' for #<RubyEventStore::Repositories::Null:0x007f9073832970>
	from (irb):15:in `method_missing'
	from (irb):28
	from /Users/rupert/.rbenv/versions/2.2.8/bin/irb:11:in `<main>'
```

Reply:
I must say I am not comfortable with this solution. It's cleaver but here is what I would expect instead

* write methods (append_to_stream, link_to_stream, delete_stream) pass and return nil
* read methods (has_event?, read_events_forward, read_all_streams_backward, etc) raise custom exception

although that does not sound quite right as well

Reply:
Technically we only need `append_to_stream` to work (do nothing, but exception-free) and everything else could rise an exception (which i prefer much more than `nil`).